shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -114,6 +114,15 @@ @implementation FlutterViewController {
   fml::scoped_nsobject<UIScrollView> _scrollView;
   fml::scoped_nsobject<UIView> _keyboardAnimationView;
   MouseState _mouseState;
+  // Timestamp after which a scroll inertia cancel event should be inferred.
+  NSTimeInterval _scrollInertiaEventStartline;
+  // When an iOS app is running in emulation on an Apple Silicon Mac, trackpad input goes through
+  // a translation layer, and events are not received with precise deltas. Due to this, we can't
```

Comment:
When running unmodified iOS app on Mac, events which would usually be `CGEvent` or `NSEvent` are turned into `UIEvent`. Since the iOS code can't be modified, there is some layer (which I believe is called `UINSMouseEventTranslator` based on debug data), that acts as a bridge between UIKit and AppKit. This is an apple private implementation detail and not documented.